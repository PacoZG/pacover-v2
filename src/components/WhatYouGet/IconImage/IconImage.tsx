import Image from 'next/image'
import React from 'react'
import { useTranslations } from 'next-intl'

type IconImageProps = {
  name: string
  imageUrl?: string,
  certUrl?: string,
  altName?: string,
  darkThemeTextColour: string,
  IconComponent?: React.ReactNode,
  skillToolTipMsg: boolean
}

const IconImage: React.FC<IconImageProps> = ({
  name,
  imageUrl,
  certUrl,
  altName,
  darkThemeTextColour,
  IconComponent,
  skillToolTipMsg
}) => {
  const t = useTranslations('skills')

  return (
    <div>
      <a
        skill-tool-tip-msg={skillToolTipMsg ? t('click-for-certificate'): undefined}
        className="skill-card"
        href={certUrl}
        target="_blank"
        rel="noopener noreferrer"
      >
        {imageUrl && altName ? (
          <Image
            className="skill-logo object-contain"
            src={imageUrl}
            alt={altName}
            width={600}
            height={600}
          />
        ) : (
          IconComponent
        )}
        <p className={`skill-text dark:${darkThemeTextColour}`}>{name}</p>
      </a>
    </div>
  )
}

export default IconImage
