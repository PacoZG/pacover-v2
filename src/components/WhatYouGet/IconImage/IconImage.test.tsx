/* eslint-disabled */
import React from 'react'
import { render } from '@testing-library/react'
import { beforeEach, describe, expect, test, vi } from 'vitest'
import IconImage from '@/components/WhatYouGet/IconImage/IconImage'

describe('IconImage', () => {
  describe('given configured with dependencies', () => {
    beforeEach(() => {
      window.scrollTo = vi.fn()

      vi.mock('next-intl', () => ({
        useTranslations: () => (key: string) => key,
      }))
    })

    describe('IconImage component', () => {
      test('matches the snapshot', () => {
        const { container } = render(
          <IconImage
            certUrl="http://some-cert-url"
            imageUrl="http://some-image-url"
            altName="some-logo"
            imageExtraStyle="some-extra-style"
            darkThemeTextColour="some-dark-theme-colour"
            skillToolTipMsg={true}
            name={'some-name'}
          />
        )

        expect(container).toMatchSnapshot()
      })
    })
  })
})
