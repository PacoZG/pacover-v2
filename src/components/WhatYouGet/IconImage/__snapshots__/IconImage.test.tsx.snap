// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconImage > given configured with dependencies > IconImage component > matches the snapshot 1`] = `
<div>
  <div>
    <a
      class="skill-card"
      href="http://some-cert-url"
      rel="noopener noreferrer"
      skill-tool-tip-msg="click-for-certificate"
      target="_blank"
    >
      <img
        alt="some-logo"
        class="skill-logo object-contain some-extra-style"
        data-nimg="1"
        decoding="async"
        height="600"
        loading="lazy"
        src="/_next/image?url=http%3A%2F%2Fsome-image-url&w=1200&q=75"
        srcset="/_next/image?url=http%3A%2F%2Fsome-image-url&w=640&q=75 1x, /_next/image?url=http%3A%2F%2Fsome-image-url&w=1200&q=75 2x"
        style="color: transparent;"
        width="600"
      />
      <p
        class="skill-text some-dark-theme-colour"
      >
        some-name
      </p>
    </a>
  </div>
</div>
`;
